### ISSUE TITLE
Stop using `anyhow::Error`?

### ISSUE THREAD

**Issue by strohel** [cameleon-rs/cameleon#183]
I see that `cameleon` uses `anyhow::Error`, but very lightly: https://github.com/search?q=repo%3Acameleon-rs%2Fcameleon%20anyhow&type=code (only for 2 `Io` error variants).

Anyhow specifically isn't designed for library errors, so it would be best to replace these with something different, perhaps custom `thiserror`-derived errors (which has a benefit of not ending up in the crate signature).

A nice side-effect would be removing the dependency (for people that don't use `anyhow` in their apps).

Or @Y-Nak do you recall having a specific reason to use `anyhow`? CC @bschwind @mbernat.

**Comment by Y-Nak** [comment]
I agree to remove `anyhow` and use `thiserror` instead. I don't remember why I decided to use `anyhow` here. I probably intended to use it as a stopgap before elaborating error kinds, then forgot to fix it 😅
